internal/deploymentcontext/deployment_context.go

```diff
@@ -25,7 +25,16 @@ import (
 	"github.com/block/ftl/internal/rpc"
 )
 
-type DeploymentContextProvider <-chan DeploymentContext
+type DeploymentContextProvider func(ctx context.Context) <-chan DeploymentContext
+
+type SecretsProvider func(ctx context.Context) map[string][]byte
+type ConfigProvider func(ctx context.Context) map[string][]byte
+
+type RouteProvider interface {
+	Subscribe() chan string
+	Unsubscribe(c chan string)
+	Route(module string) string
+}
```

Comment:
These abstractions (secrets+config+routing) seem unnecessary given that that's the bulk of the DeploymentContext? Each environment has completely different mechanisms, so can we just have a single DeploymentContextProvider implementation per-environment without sub-abstractions?